### DISCUSSION TITLE
LAMMPS + Allegro Force Output is NaN

### DISCUSSION THREAD

**Discussion by ryandeng1** [mir-group/allegro#53] [Q&A]
I have question about the Allegro model when using it with LAMMPS. I noticed that in LAMMPS, there are several atoms that are of the same tag or global atom ID, particularly when the atom is close to the periodic boundary. Why are multiple copies of the atom needed when using LAMMPS or Allegro and why does Allegro output NaN as a force if I don't do whatever LAMMPS does?

Currently, I don't replicate atoms near periodic boundaries and I am observing this behavior with atoms whose neighbors cross the periodic boundary. 

For example, if the periodic boundaries are [100, 100, 100] in 3 dimensions and I have an atom A with position (1, 1, 1), in my implementation its neighbor N will have position (-1, -1, -1) or something like that instead of (99, 99, 99).

However, in LAMMPS I observe that there will be atoms for example with position (102, 102, 102) connected to a "copy" of the target atom A at and this copy of A will have position (101, 101, 101) and there might even be multiple copies of A, all with the same position (101, 101, 101), which is equivalent to the original position of A if we incorporate periodic boundaries.

For these atoms such as atom A I mentioned in the example, the force output for them will be NaN. I have checked that the inputs are otherwise the same with respect to the atom A. I am wondering why this might be the case?

Hope my example makes some sense and thanks in advance for your help!

**Comment by Linux-cpp-lisp** [comment]
Hi @ryandeng1 ,

Thanks for your interest in our work!

I'm a little confused by the question--- you are running a simulation in LAMMPS with Allegro and getting NaN?  Or your simulations with Allegro in LAMMPS are fine, and you are observing NaN somewhere else?

**Comment by ryandeng1** [comment]
Hi, sorry. I managed to fix the NaN issue. My question is with LAMMPS. Why is it that there are multiple indices in the atom array list with the same tag when running Allegro? The multiple indices have different absolute locations, but the same location when considering the periodic boundary.

Is this something special used by Allegro or just inherent within LAMMPS? Does Allegro expect the input to look like this when feeding it in within LAMMPS?
